src/argilla/server/policies.py

```diff
@@ -64,6 +65,20 @@ def delete(cls, user: User) -> PolicyAction:
         return lambda actor: actor.role == UserRole.admin
 
 
+class DatasetPolicy:
+    @classmethod
+    def get(cls, workspace: Workspace) -> PolicyAction:
+        return lambda actor: actor.role == UserRole.admin or workspace in actor.workspaces
+
+    @classmethod
+    def create(cls, workspace: Workspace) -> PolicyAction:
+        return lambda actor: actor.role == UserRole.admin or workspace in actor.workspaces
+
+    @classmethod
+    def delete(cls, dataset: Dataset) -> PolicyAction:
+        return lambda actor: actor.role == UserRole.admin or actor.username == dataset.created_by
```

Comment:
If you are really getting an ORM on the handler then you can do `actor.is_admin`